### PR TITLE
Platform ops

### PR DESCRIPTION
**Pull request by APorumb**
Suggestion no. 1 of how driver files should be organized/renamed.

-moved drivers api from drivers/driver_type to drivers/api

### CODE REVIEW

**Comment by adrimbarean** [comment]
For the CI errors try rebasing the branch on master and doing a build test. There were some changes since the branch was created for some of the projects. Look for projects ad463x_fmcz (iio), ad7124-8pmdz (iio_uart), ad7124-8pmdz (iio_wifi), ada4250_ardz (simple), iio_aducm3029 (iio) and aducm_blinky_example (simple). Their src.mk should be changed.

Does the build server automatically rebase PR branches on master? The errors it's throwing seem to point to that.

**Comment by mchindri** [comment]
> For the CI errors try rebasing the branch on master and doing a build test. There were some changes since the branch was created for some of the projects. Look for projects ad463x_fmcz (iio), ad7124-8pmdz (iio_uart), ad7124-8pmdz (iio_wifi), ada4250_ardz (simple), iio_aducm3029 (iio) and aducm_blinky_example (simple). Their src.mk should be changed.
> 
> Does the build server automatically rebase PR branches on master? The errors it's throwing seem to point to that.

I think this PR is to suggest a new approach. I wouldn't change projects yet, only when we are sure the changes are ok.

**Comment by adrimbarean** [comment]
> > For the CI errors try rebasing the branch on master and doing a build test. There were some changes since the branch was created for some of the projects. Look for projects ad463x_fmcz (iio), ad7124-8pmdz (iio_uart), ad7124-8pmdz (iio_wifi), ada4250_ardz (simple), iio_aducm3029 (iio) and aducm_blinky_example (simple). Their src.mk should be changed.
> > Does the build server automatically rebase PR branches on master? The errors it's throwing seem to point to that.
> 
> I think this PR is to suggest a new approach. I wouldn't change projects yet, only when we are sure the changes are ok.

Should also mark it as draft then. I remember there was the question about why the CI fails.

File: projects/ada4250_ardz/src.mk [hunk 1]
```diff
@@ -3,19 +3,28 @@
 SRC_DIRS += $(PROJECT)/src
 SRC_DIRS += $(DRIVERS)/amplifiers/ada4250
 SRC_DIRS += $(NO-OS)/iio/iio_app
-SRC_DIRS += $(NO-OS)/drivers/irq
-SRC_DIRS += $(NO-OS)/drivers/gpio
 
-SRC_DIRS += $(PLATFORM_DRIVERS)
```
Comment (mchindri) [inline projects/ada4250_ardz/src.mk:9]: what was wrong with this line?

File: projects/iio_aducm3029/src.mk [hunk 2]
```diff
@@ -10,6 +10,5 @@ SRC_DIRS += $(PROJECT)/src \
 
 ifeq '$(TINYIIOD)' 'y'
 SRC_DIRS += $(NO-OS)/iio/iio_app
-SRC_DIRS += $(NO-OS)/drivers/irq
-SRC_DIRS += $(NO-OS)/drivers/gpio
+SRC_DIRS += $(NO-OS)/api
```
Comment (mchindri) [inline projects/iio_aducm3029/src.mk:13]: Why not use SRC_DIRS += $(NO-OS)/api everywhere?

Comment (APorumb) [reply projects/iio_aducm3029/src.mk:13]: As I discussed with @amiclaus , we should better include only the used files and not the whole folder. In this way, the user will know exactly which files are used by the project.

File: projects/ada4250_ardz/src.mk (hunk 1 above)
Comment (APorumb) [reply projects/ada4250_ardz/src.mk:9]: As I mentioned in the other comment, I didn't want to add the whole folder (which includes also unused files).

**Comment by buha** [comment]
Looks good but it seems you've missed some changes in one project:
```
[dari@plex no-OS2]$ ag -Q "spi/spi.c"
projects/ad4110/src.mk
13:SRCS += $(DRIVERS)/spi/spi.c						\
[dari@plex no-OS2]$ ag -Q "gpio/gpio.c"
projects/ad4110/src.mk
15:	$(DRIVERS)/gpio/gpio.c						\
[dari@plex no-OS2]$ ag -Q "i2c/i2c.c"
[dari@plex no-OS2]$ ag -Q "uart/uart.c"
[dari@plex no-OS2]$ ag -Q "irq/irq.c"
projects/ad4110/src.mk
14:	$(DRIVERS)/irq/irq.c						\
[dari@plex no-OS2]$ ag -Q "tdm/tdm.c"
```

**Comment by buha** [comment]
And since you're at it, add a second commit to add a builds.json file for ad4110, this way, the build system will catch mishaps in the future:
https://github.com/analogdevicesinc/no-OS/tree/master/projects/ad4110

**Comment by APorumb** [comment]
> Looks good but it seems you've missed some changes in one project:
> 
> ```
> [dari@plex no-OS2]$ ag -Q "spi/spi.c"
> projects/ad4110/src.mk
> 13:SRCS += $(DRIVERS)/spi/spi.c						\
> [dari@plex no-OS2]$ ag -Q "gpio/gpio.c"
> projects/ad4110/src.mk
> 15:	$(DRIVERS)/gpio/gpio.c						\
> [dari@plex no-OS2]$ ag -Q "i2c/i2c.c"
> [dari@plex no-OS2]$ ag -Q "uart/uart.c"
> [dari@plex no-OS2]$ ag -Q "irq/irq.c"
> projects/ad4110/src.mk
> 14:	$(DRIVERS)/irq/irq.c						\
> [dari@plex no-OS2]$ ag -Q "tdm/tdm.c"
> ```

Json file could not be added due to missing hardware hdl on master.

**Review by buha — APPROVED** [review]

**Review by amiclaus — COMMENTED** [review]
Looks like some conflicts between your branch and the master branch have appeared in the meantime. I think it has to do with https://github.com/analogdevicesinc/no-OS/commit/07f34a2c0a908d0b6d0ca6312fd8f0abecbc4612 .

**Comment by APorumb** [comment]
> Looks like some conflicts between your branch and the master branch have appeared in the meantime. I think it has to do with [07f34a2](https://github.com/analogdevicesinc/no-OS/commit/07f34a2c0a908d0b6d0ca6312fd8f0abecbc4612) .

Conflicts solved.

File: projects/adxrs290-pmdz/src.mk [hunk 3]
```diff
@@ -24,9 +24,9 @@ SRC_DIRS += $(PLATFORM_DRIVERS)
...
 
-SRCS +=	$(DRIVERS)/irq/irq.c \
-	$(DRIVERS)/gpio/gpio.c \
+SRCS +=	$(DRIVERS)/api/irq.c \
+	$(DRIVERS)/api/gpio.c \
 	$(NO-OS)/util/list.c \
 	$(NO-OS)/util/fifo.c \
 	$(NO-OS)/util/util.c \
-	$(DRIVERS)/spi/spi.c
+	$(DRIVERS)/api/spi.c
```
Comment (amiclaus) [inline projects/adxrs290-pmdz/src.mk:32]: nit: new line missing in the new patch.

**Review by amiclaus — APPROVED** [review]
lgtm, one more nit on my side

**Pull request merged**